### PR TITLE
fix parsing yul expressions

### PR DESCRIPTION
**Pull request by OmarTawfik**

### CODE REVIEW

File: crates/solidity/inputs/schema/grammar/topics/06-yul/03-yul-expressions.yml [hunk 1]
```diff
@@ -4,10 +4,27 @@
...
-      - reference: "YulIdentifierPath"
       - reference: "YulFunctionCall"
       - reference: "YulLiteral"
 
+- name: "YulFunctionCall"
+  kind: "Rule"
+  unversioned:
+    sequence:
+      - reference: "YulIdentifierPath"
+      # TODO: temporary hack until precedence is implemented properly here.
```
Comment (OmarTawfik) [inline crates/solidity/inputs/schema/grammar/topics/06-yul/03-yul-expressions.yml:15]: Noting the temporary hack until precedence is redesigned/fixed.

Comment (AntonyBlakey) [reply crates/solidity/inputs/schema/grammar/topics/06-yul/03-yul-expressions.yml:15]: Precedence is working AFAIK, so can you explain what the issue is here?

Comment (OmarTawfik) [reply crates/solidity/inputs/schema/grammar/topics/06-yul/03-yul-expressions.yml:15]: This is the failure I mentioned earlier. The current PR is a hack to make parsing succeed, while we investigate next.

Below is the precedence alternative. It currently fails with `A TerminalTrie with no entries should never have been constructed`. It assumes the existence of a binary operator, and fails as such. Might also be a part of our discussion on enforcing the explicit structure of each operator type (ternary/binary/unary) via YAML (an EBNF variant) instead detecting it later.

```yml
- name: "YulExpression"
  kind: "Rule"
  unversioned:
    parserType: "Precedence"
    choice:
      - reference: "YulFunctionCall"
      - reference: "YulPrimaryExpression"

- name: "YulFunctionCall"
  kind: "Rule"
  unversioned:
    sequence:
      - reference: "YulExpression"
      - delimitedBy:
          open: "("
          expression:
            optional:
              separatedBy:
                separator: ","
                expression:
                  reference: "YulExpression"
          close: ")"

- name: "YulPrimaryExpression"
  kind: "Rule"
  unversioned:
    choice:
      - reference: "YulIdentifierPath"
      - reference: "YulLiteral"
```

Comment (AntonyBlakey) [reply crates/solidity/inputs/schema/grammar/topics/06-yul/03-yul-expressions.yml:15]: Ah, I get it. easy to fix. I'll approve and schedule a fix.

**Review by AntonyBlakey — APPROVED** [review]

**Pull request merged**